### PR TITLE
Add `resource_client_file_checks` setting to disable corrupt PNG/TXD/DFF warnings

### LINKED ISSUE #2541: Remove/change DFF file warnings
Some dff files will output a warning to the server log like this:
`WARNING: File 'file_path.dff' in resource 'resource_name' contains errors.`

I suggest either these warnings are removed entirely or that there is a way to turn them off via a mtaserver.conf setting.

Why: these warnings are wrong/irrelevant for many models that work perfectly fine as far as I've tested.

### PR DESCRIPTION
**Pull request by Fernando-A-Rocha**
Fixes #2541 by adding a new mtaserver.conf setting `resource_client_file_checks` (enabled '1' by default), which allows the server to not check files in resources that end in PNG, DFF and TXD for errors. This would previously output a lot of unwanted annoying warnings for certain files, especially when they were encrypted.

### CODE REVIEW

File: Server/mods/deathmatch/logic/CMainConfig.h [hunk 1]
```diff
@@ -227,4 +228,5 @@ class CMainConfig : public CXMLConfig
     int                        m_bFakeLagCommandEnabled;
     int                        m_iPlayerTriggeredEventIntervalMs;
     int                        m_iMaxPlayerTriggeredEventsPerInterval;
+    int                        m_bCheckResourceClientFilesEnabled;
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/CMainConfig.h]: bool?

File: Server/mods/deathmatch/logic/CMainConfig.h [hunk 2]
```diff
@@ -126,6 +126,7 @@ class CMainConfig : public CXMLConfig
     const std::vector<SString>& GetOwnerEmailAddressList() const { return m_OwnerEmailAddressList; }
     bool                        IsDatabaseCredentialsProtectionEnabled() const { return m_bDatabaseCredentialsProtectionEnabled != 0; }
     bool                        IsFakeLagCommandEnabled() const { return m_bFakeLagCommandEnabled != 0; }
+    bool                        IsCheckResourceClientFilesEnabled() const { return m_bCheckResourceClientFilesEnabled != 0; }
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/CMainConfig.h]: `noexcept` + remove `!= 0`

**Review by TracerDS — CHANGES_REQUESTED** [review]

File: Server/mods/deathmatch/logic/CMainConfig.h (hunk 1 above)
Comment (Fernando-A-Rocha) [reply Server/mods/deathmatch/logic/CMainConfig.h:231]: > bool?

won't work with GetIntSettingList

Comment (Fernando-A-Rocha) [reply Server/mods/deathmatch/logic/CMainConfig.h:231]: Let me refactor

Comment (Fernando-A-Rocha) [reply Server/mods/deathmatch/logic/CMainConfig.h:231]: look at m_bDatabaseCredentialsProtectionEnabled

**Comment by Fernando-A-Rocha** [comment]
I think it's alright now

File: Server/mods/deathmatch/logic/CMainConfig.cpp [hunk 3]
```diff
@@ -1465,6 +1465,7 @@ const std::vector<SIntSetting>& CMainConfig::GetIntSettingList()
         {false, false, 0, 0, 1, "fakelag", &m_bFakeLagCommandEnabled, NULL},
         {true, true, 50, 1000, 5000, "player_triggered_event_interval", &m_iPlayerTriggeredEventIntervalMs, &CMainConfig::OnPlayerTriggeredEventIntervalChange},
         {true, true, 1, 100, 1000, "max_player_triggered_events_per_interval", &m_iMaxPlayerTriggeredEventsPerInterval, &CMainConfig::OnPlayerTriggeredEventIntervalChange},
+        {true, true, 0, 1, 1, "resource_client_file_checks", &m_checkResourceClientFiles, NULL},
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/CMainConfig.cpp]: `nullptr` instead of `NULL`

**Review by TracerDS — CHANGES_REQUESTED** [review]

**Review by TracerDS — APPROVED** [review]

**Comment by Fernando-A-Rocha** [comment]
Tbh if the majority agrees, i can just remove the png/txd/dff checks coz they dont make much sense. It's things you're meant to do in Lua, and functions that take png and dff/txd files already do checking.

File: Server/mods/deathmatch/logic/CMainConfig.h [hunk 4]
```diff
@@ -227,4 +228,5 @@ class CMainConfig : public CXMLConfig
     int                        m_bFakeLagCommandEnabled;
     int                        m_iPlayerTriggeredEventIntervalMs;
     int                        m_iMaxPlayerTriggeredEventsPerInterval;
+    int                        m_checkResourceClientFiles;
```
Comment (TheNormalnij) [inline Server/mods/deathmatch/logic/CMainConfig.h:231]: How does MTA handle the situation when `resource_client_file_checks` is missing in mtaserver.conf?

Comment (Fernando-A-Rocha) [reply Server/mods/deathmatch/logic/CMainConfig.h:231]: It defaults to 1 (the default value defined). I tested.

https://github.com/multitheftauto/mtasa-blue/blob/e92702ca56a30b0c434cc29402b2d378ba4d7da4/Server/mods/deathmatch/logic/CMainConfig.cpp#L512-L520

**Review by TheNormalnij — APPROVED** [review]

**Comment by TheNormalnij** [comment]
@Fernando-A-Rocha documentation

**Comment by Fernando-A-Rocha** [comment]
> @Fernando-A-Rocha documentation

Done
https://wiki.multitheftauto.com/wiki/Server_mtaserver.conf#resource_client_file_checks

**Pull request merged**